### PR TITLE
Allow merging config with actual default config

### PR DESCRIPTION
**Pull request by bew**
Trying to use this plugin and configure it is really a pain without this behavior.

I removed a comment about this, that also mentioned the wish to always merge config with last config, but I don't really see this as useful in any way..
Merging with real default config is always useful though 👀

### CODE REVIEW

File: lua/neorepl/init.lua [hunk 1]
```diff
@@ -121,23 +121,22 @@ package.loaded['neorepl.bufs'] = bufs
...
+local current_config = vim.deepcopy(DEFAULT_CONFIG)
+
 ---Set default configuration
 ---@param config? neorepl.Config
 function neorepl.config(config)
-  -- TODO: merge with the old one.
-  -- to do so, we need a way of resetting values back to default,
-  -- in particular on_init, env_lua, lang. either false or have neorepl.DEFAULT constant
-  config = validate(config)
+  config = validate(config or {})
```
Comment (ii14) [inline lua/neorepl/init.lua]: `validate` already does `or {}` internally? Am I missing something?

**Review by ii14 — COMMENTED** [review]
If I remember correctly, the problem there was that there is no way of removing existing on_init or env_lua back to nil, because on tables nil is as if the value doesn't exist at all. Maybe I was overthinking it though.

File: lua/neorepl/init.lua (hunk 1 above)
Comment (bew) [reply lua/neorepl/init.lua]: Ah yes, didn't see it, will change that

**Comment by bew** [comment]
> no way of removing existing on_init or env_lua back to nil

The simple solution to something like that is to not set them to nil in the first place, but set them to placeholder functions (that are easy to replicate if possible, to ease configuration or making custom ones):
- `on_init` can be a simple `function(bufnr) end`
- `env_lua` can be `function() return _G end` or similar 🤔

Tell me if you want me to impl this in the PR or separate ones

**Comment by ii14** [comment]
Well, in that case I guess that's what you can do to reset them, without using `nil`. I guess that's fine.

Thanks for the PR.

**Comment by bew** [comment]
Also true 👍

**Pull request merged**